src/overlays/actors/ovl_En_Ru/z_en_ru.c

```diff
@@ -76,34 +86,285 @@ static DamageTable D_80A39488 = {
+        Math_SmoothStepToS(&this->trackTarget.x, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->trackTarget.y, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->headRot.x, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->headRot.y, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->torsoRot.x, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->torsoRot.y, 0, 4, 0x3E8, 1);
+    }
+
+    EnRu_UpdateEyes(this, 3);
+    EnRu_PlayWalkingSound(this, play);
+    SubS_FillLimbRotTables(play, this->limbRotTableY, this->limbRotTableZ, RU2_LIMB_MAX);
+}
+
+void EnRu_DoNothing(EnRu* this, PlayState* play) {
+}
+
+void EnRu_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnRu* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gAdultRutoSkel, NULL, this->jointTable, this->morphTable, RU2_LIMB_MAX);
+    EnRu_ChangeAnimation(&this->skelAnime, 0);
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+
+    // path is prepared but unused after this
+    this->path = SubS_GetPathByIndex(play, RU_GET_PATH(thisx), 0x3F);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = EnRu_DoNothing;
+    this->actor.gravity = -4.0f;
+}
+
+void EnRu_Destroy(Actor* thisx, PlayState* play) {
+    EnRu* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void EnRu_Update(Actor* thisx, PlayState* play) {
+    EnRu* this = THIS;
+
+    this->actionFunc(this, play);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    EnRu_UpdateModel(this, play);
+    EnRu_CollisionCheck(this, play);
+}
+
+s32 EnRu_OverrideLimbdraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx,
+                          Gfx** gfx) {
+    EnRu* this = THIS;
+
+    if (limbIndex == RU2_LIMB_HEAD) {
+        Matrix_Translate(1500.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+        Matrix_RotateXS(this->headRot.y, MTXMODE_APPLY);
+        Matrix_RotateZS(this->headRot.x, MTXMODE_APPLY);
+        Matrix_Translate(-1500.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+    }
+
+    if (limbIndex == RU2_LIMB_TORSO) {
+        Matrix_RotateYS(this->torsoRot.y, MTXMODE_APPLY);
+        Matrix_RotateXS(this->torsoRot.x, MTXMODE_APPLY);
+    }
+
+    if ((limbIndex == RU2_LIMB_TORSO) || (limbIndex == RU2_LIMB_LEFT_UPPER_ARM) ||
+        (limbIndex == RU2_LIMB_RIGHT_UPPER_ARM)) {
+        rot->y += (s32)(s16)(Math_SinS(this->limbRotTableY[limbIndex]) * 200.0f);
+        rot->z += (s32)(s16)(Math_CosS(this->limbRotTableZ[limbIndex]) * 200.0f);
+    }
+
+    return false;
+}
+
+void EnRu_PostLimbdraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx, Gfx** gfx) {
+    EnRu* this = THIS;
+    Vec3f headFocus = sHeadFocusPos;
+    Vec3f bodyPartPos = sBodyFocusPos;
```

Comment:
This data is probably actually in-function non-static, then?

Reply:
Don't think so?
Looking at the asm it looks to be making a copy.

![Screenshot_2022-08-20_19-55-33](https://user-images.githubusercontent.com/42048411/185773325-5d74d4c9-d3ad-4891-a1e1-463337b157c8.png)

Reply:
What did you do? Probably simplest to just scratch it.

Reply:
Hmm... if I do non-static in-function the data order is backwards, but if I change the order in-function the stack is wrong
![Screenshot_2022-08-20_20-03-27](https://user-images.githubusercontent.com/42048411/185773510-b36703a6-3b02-44d8-9480-996c49bf600a.png)

Reply:
Did you remember to move the `sTransparencyDlist` to below that function? Cause in the diff it looks like that DL is appearing before the other data.